tensorflow_addons/metrics/matthews_correlation_coefficient.py

```diff
@@ -69,7 +69,7 @@ def __init__(
     ):
         """Creates a Matthews Correlation Coefficient instance."""
         super().__init__(name=name, dtype=dtype)
-        self.num_classes = num_classes
+        self.num_classes = max(2, num_classes)
         self.conf_mtx = self.add_weight(
             "conf_mtx",
             shape=(self.num_classes, self.num_classes),
```

Comment:
Instead of ```max(2, num_classes)```, add a check to detect if num_classes is less than 2 and appropriately throw error. See cohen_kappa.py for this.